src/vulkan-renderer/wrappers/instance.cpp

```diff
@@ -0,0 +1,174 @@
+                   const std::uint32_t engine_version, const std::uint32_t vulkan_api_version,
+                   std::optional<std::vector<std::string>> requested_instance_extensions, std::optional<std::vector<std::string>> requested_instance_layers,
+                   bool enable_validation_layers, bool enable_renderdoc_instance_layer) {
+    assert(!application_name.empty());
+    assert(!engine_name.empty());
+
+    // In Vulkan API we can use VK_MAKE_VERSION() macro to create an std::uint32_t value from major, minor and patch number.
+
+    // Get the major, minor and patch version of the application.
+    std::uint32_t app_major = VK_VERSION_MAJOR(application_version);
+    std::uint32_t app_minor = VK_VERSION_MINOR(application_version);
+    std::uint32_t app_patch = VK_VERSION_PATCH(application_version);
+
+    // Get the major, minor and patch version of the engine.
+    std::uint32_t engine_major = VK_VERSION_MAJOR(engine_version);
+    std::uint32_t engine_minor = VK_VERSION_MINOR(engine_version);
+    std::uint32_t engine_patch = VK_VERSION_PATCH(engine_version);
+
+    // Get the major, minor and patch version of the requested Vulkan API interface.
+    std::uint32_t vulkan_api_major = VK_VERSION_PATCH(vulkan_api_version);
+    std::uint32_t vulkan_api_minor = VK_VERSION_PATCH(vulkan_api_version);
+    std::uint32_t vulkan_api_patch = VK_VERSION_PATCH(vulkan_api_version);
+
+    spdlog::debug("Initialising Vulkan instance.");
+    spdlog::debug("Application name: '{}'", application_name.c_str());
+    spdlog::debug("Application version: {}.{}.{}", app_major, app_minor, app_patch);
+    spdlog::debug("Engine name: '{}'", engine_name.c_str());
+    spdlog::debug("Engine version: {}.{}.{}", engine_major, engine_minor, engine_patch);
+    spdlog::debug("Requested Vulkan API version: {}.{}.{}", vulkan_api_patch, vulkan_api_minor, vulkan_api_patch);
+
+    VkApplicationInfo app_info = {};
+    app_info.sType = VK_STRUCTURE_TYPE_APPLICATION_INFO;
+    app_info.pNext = nullptr;
+    app_info.pApplicationName = application_name.c_str();
+    app_info.applicationVersion = application_version;
+    app_info.pEngineName = engine_name.c_str();
+    app_info.engineVersion = engine_version;
+    app_info.apiVersion = vulkan_api_version;
+
+    std::vector<const char *> instance_extension_wishlist = {
+#ifndef NDEBUG
+        // In debug mode, we use the following instance extensions:
+        // This one is for assigning internal names to Vulkan resources.
+        VK_EXT_DEBUG_UTILS_EXTENSION_NAME,
+        // This one is for setting up a Vulkan debug report callback function.
+        VK_EXT_DEBUG_REPORT_EXTENSION_NAME,
+#endif
+    };
+
+    std::uint32_t number_of_glfw_extensions = 0;
+
+    // Because this requires some dynamic libraries to be loaded, this may take even up to some seconds!
+    auto glfw_extensions = glfwGetRequiredInstanceExtensions(&number_of_glfw_extensions);
+
+    if (number_of_glfw_extensions == 0) {
```

Comment:
Can be replaced with `if (!number_of_glfw_extensions)`. Not sure if our clang-tidy allows this though.

Reply:
I renamed it to `glfw_extension_count`.

Reply:
I would like to stick to the `if(value == 0)` for numeric variables.